crates/sui-indexer/src/models/transactions.rs

```diff
@@ -273,26 +273,27 @@ pub async fn tx_events_to_sui_tx_events(
                 "Failed to convert to sui event with Error: {e}",
             ))
         })?;
-    let event_move_struct_layouts = event_move_type_layouts
+    let event_move_datatype_layouts = event_move_type_layouts
         .into_iter()
         .filter_map(|move_type_layout| match move_type_layout {
-            MoveTypeLayout::Struct(s) => Some(s),
+            MoveTypeLayout::Struct(s) => Some(MoveDatatypeLayout::Struct(s)),
+            MoveTypeLayout::Enum(e) => Some(MoveDatatypeLayout::Enum(e)),
```

Comment:
Chatted a bit about this offline but this isn't the case today, and I will look into making a followup PR with this change since it will be relatively high-touch across the codebase.